stage1_fly/run/main.go

```
@@ -0,0 +1,381 @@
+	}
+
+	// Merge command-line Volumes with ImageManifest's MountPoints
+	for _, mp := range imApp.MountPoints {
+		// Check if we have a volume for this mountpoint
+		tuple, exists := namedVolumeMounts[mp.Name]
+		if !exists || tuple.V.Name == "" {
+			log.Fatalf("fly: missing volume for mountpoint %q", mp.Name)
+		}
+
+		// If empty, fill in ReadOnly bit
+		if tuple.V.ReadOnly == nil {
+			v := tuple.V
+			v.ReadOnly = &mp.ReadOnly
+			namedVolumeMounts[mp.Name] = volumeMountTuple{M: tuple.M, V: v}
+			log.Printf("Adding %+v", namedVolumeMounts[mp.Name])
+		}
+	}
+
+	// Gather host mounts which we make MS_SHARED if passed as a volume source
+	hostMounts, err := getHostMounts()
+	if err != nil {
+		log.Fatalf("fly: error gathering host mounts")
+	}
+
+	argFlyMounts := []flyMount{}
+	var flags uintptr = syscall.MS_BIND // TODO: allow optional | syscall.MS_REC
+	for _, tuple := range namedVolumeMounts {
+		if _, isHostMount := hostMounts[tuple.V.Source]; isHostMount {
+			// Mark the host mount as SHARED so the container's changes to the mount are propagated to the host
+			argFlyMounts = append(argFlyMounts,
+				flyMount{"", "", tuple.V.Source, "none", syscall.MS_REC | syscall.MS_SHARED},
+			)
+		}
+		argFlyMounts = append(argFlyMounts,
+			flyMount{tuple.V.Source, rfs, tuple.M.Path, "none", flags},
+		)
+
+		if tuple.V.ReadOnly != nil && *tuple.V.ReadOnly {
+			argFlyMounts = append(argFlyMounts,
+				flyMount{"", rfs, tuple.M.Path, "none", flags | syscall.MS_REMOUNT | syscall.MS_RDONLY},
+			)
+		}
+	}
+	return argFlyMounts, nil
+}
+
+func stage1() int {
+	uuid, err := types.NewUUID(flag.Arg(0))
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "UUID is missing or malformed")
+		return 1
+	}
+
+	root := "."
+	p, err := stage1commontypes.LoadPod(root, uuid)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to load pod: %v\n", err)
+		return 1
+	}
+
+	if len(p.Manifest.Apps) != 1 {
+		log.Fatalf("Flavor %q only supports 1 application per Pod for now.", flavor)
```

Ditto. And maybe return an error instead like you seem to be doing above?